### PR TITLE
Audio playrate

### PR DESCRIPTION
**Pull request by mikecoker**
Adding playback rate to audio players and audio source to allow the user to create pitch modifications to sounds.

Changelog:
 *  added playback rate to audio source
 * added playback rate to mini-game web player
 * added playback rate to both players for other web platforms
 * added a stub for playback rate to native, though it is unsupported at this time

### CODE REVIEW

File: cocos/audio/audio-source.ts [hunk 1]
```diff
@@ -197,6 +199,30 @@ export class AudioSource extends Component {
...
+    /**
+     * @en
+     * The playback rate of this audio source (0.0 to 10.0).<br>
+     * Note: Playback rate may be ineffective on some platforms.
+     * @zh
+     * 此音频源的播放速率（0.0 到 10.0）。<br>
+     * 注意：播放速率在某些平台上可能无效。<br>
+     */
+    @range([0.0, 10.0])
+    @tooltip('i18n:audio.playbackrate')
```
Comment (PPpro) [inline cocos/audio/audio-source.ts:211]: please add the localization in the file `editor/i18n/zh/localization.js` and `editor/i18n/en/localization.js`

File: cocos/audio/audio-source.ts [hunk 2]
```diff
@@ -197,6 +199,30 @@ export class AudioSource extends Component {
...
+    set playbackRate (val: number) {
+        if (Number.isNaN(val)) { console.warn('illegal playback rate!'); return; }
```
Comment (PPpro) [inline cocos/audio/audio-source.ts:213]: We don’t want to help developers verify whether this value is `NaN`
__TypeScript__ should tell developers that they are passing an `NaN` to the `playbackRate` property,  which is a wrong approach

We hope to let the language server do the type checking job

File: cocos/audio/audio-source.ts [hunk 3]
```diff
@@ -197,6 +199,30 @@ export class AudioSource extends Component {
         return this._volume;
     }
 
+    /**
+     * @en
+     * The playback rate of this audio source (0.0 to 10.0).<br>
+     * Note: Playback rate may be ineffective on some platforms.
```
Comment (PPpro) [inline cocos/audio/audio-source.ts:205]:
```suggestion
     * Note: Playback rate is only supported on the platform Web and ByteDance for now.
```

File: cocos/audio/audio-source.ts [hunk 4]
```diff
@@ -197,6 +199,30 @@ export class AudioSource extends Component {
...
+     * @zh
+     * 此音频源的播放速率（0.0 到 10.0）。<br>
+     * 注意：播放速率在某些平台上可能无效。<br>
```
Comment (PPpro) [inline cocos/audio/audio-source.ts:208]:
```suggestion
     * 注意：播放速率目前只在 Web 和 字节 平台支持。
```

**Review by PPpro — CHANGES_REQUESTED** [review]

**Comment by PPpro** [comment]
I suggest to merge this Pull Request into the branch `v3.5.0`  @pandamicro

**Comment by pandamicro** [comment]
I rebased it to v3.5 but it seems the merge commit is causing conflict

**Comment by PPpro** [comment]
Sorry, playback rate is not currently a cross-platform feature, 
we plan to support this feature on other platforms too, this is the tracking issue: https://github.com/cocos/cocos-engine/issues/10545

I am gonna close this PR for now, but this PR will be merged after this issue is finished.
Thanks again for you contribution

**Pull request closed without merging**